x-pack/plugins/security_solution/scripts/endpoint/agent_downloader_cli/agent_downloader.ts

```diff
@@ -11,21 +11,63 @@ import type { ToolingLog } from '@kbn/tooling-log';
 import { getAgentDownloadUrl, getAgentFileName } from '../common/fleet_services';
 import { downloadAndStoreAgent } from '../common/agent_downloads_service';
 
+// Decrement the patch version by 1
+const decrementPatchVersion = (version: string): string => {
+  const [major, minor, patch] = version.split('.').map(Number);
```

Comment:
can you see if the `semver` package can help here. The code here assumes that `version` is a simple version of three separated values by `.` which may not be true.

Also - what happens if `patch` is `0`? Example: `8.15.0`, `9.0.0`, etc...